start_simulation.sh

```diff
@@ -3,4 +3,64 @@
 SCRIPT=$(realpath "$0")
 SCRIPTDIR=$(dirname "${SCRIPT}")
 
-(cd ${SCRIPTDIR} && ${SCRIPTDIR}/tools/bazelisk run //spines:bullet_spine -- --show)
+URL_ARCHIVE="https://github.com/upkie/upkie/releases/download"
+
+if [ ! -f docs/Doxyfile ]; then
+    echo "Unable to find file 'docs/Doxyfile' for version number";
+    exit;
+fi
+# no v at start of version number here
+VERSION=$(awk '/^PROJECT_NUMBER/{print $3}' docs/Doxyfile)
+
+SYSTEM=$(uname -s)
+ARCH=$(uname -m)
+
+if [[ "$SYSTEM" == Darwin ]]; then
+    echo "MacOS system"
+    if [[ "$ARCH" == x86_64* ]]; then
+        echo "X64 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_x86_bullet_spine.tar.gz
+    elif [[ "$ARCH" == i*86 ]]; then
+        echo "X32 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_x86_bullet_spine.tar.gz
+    elif  [[ "$ARCH" == arm* ]]; then
+        echo "ARM Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_arm64_bullet_spine.tar.gz
+    else
+        echo "Unsupported architecture $ARCH"
+    fi
+elif  [[ "$SYSTEM" == Linux ]]; then
+    echo "Linux system"
+    if [[ "$ARCH" == x86_64* ]]; then
+        echo "X64 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/linux_amd64_bullet_spine.tar.gz
+    else
+        echo "Unsupported architecture: $ARCH"
+    fi
+else
+    echo "Unsupported system: $SYSTEM"
+fi
+
+if [[ -z "$SPINE_ARCHIVE" ]]; then
+    echo "Building the simulation spine locally...";
+    (cd ${SCRIPTDIR} && ${SCRIPTDIR}/tools/bazelisk run //spines:bullet_spine -- --show)
+else
+    echo "Downloading the simulation spine from $SPINE_ARCHIVE..."
+    tmp_dir=$(mktemp -d)
+    pushd $tmp_dir
```

Comment:
This works. There is _one more thing™_ I didn't pay attention to in the previous review: if we download and uncompress to a temporary directory every time, then every time a user downloads the script they will have to wait for a full download/decompression cycle.

How about we `mkdir` a `cache/` directory directly here, and add it to `.gitignore` so that it doesn't mix with source code?

Reply:
Pushed solution  1 proposal

Reply:
Solution 1 sounds better to me too :+1: But I overlooked indeed the fact that the binary can be updated. We can address that as a follow up.